Sources/SwiftIfConfig/StaticBuildConfiguration.swift

```diff
@@ -0,0 +1,402 @@
+@_spi(ExperimentalLanguageFeatures) import SwiftParser
+import SwiftSyntax
+#endif
+
+/// A statically-determined build configuration that can be used with any
+/// API that requires a build configuration. Static build configurations can
+/// be (de-)serialized via Codable.
+///
+/// Static build configurations can not be used for canImport checks, because
+/// such checks require deeper integration with the compiler itself.
+public struct StaticBuildConfiguration: Codable {
+  public init(
+    customConditions: Set<String> = [],
+    features: Set<String> = [],
+    attributes: Set<String> = [],
+    targetOSNames: Set<String> = [],
+    targetArchitectures: Set<String> = [],
+    targetEnvironments: Set<String> = [],
+    targetRuntimes: Set<String> = [],
+    targetPointerAuthenticationSchemes: Set<String> = [],
+    targetObjectFileFormats: Set<String> = [],
+    targetPointerBitWidth: Int = 64,
+    targetAtomicBitWidths: [Int] = [],
+    endianness: Endianness = .little,
+    languageVersion: VersionTuple,
+    compilerVersion: VersionTuple
+  ) {
+    self.customConditions = customConditions
+    self.features = features
+    self.attributes = attributes
+    self.targetOSNames = targetOSNames
+    self.targetArchitectures = targetArchitectures
+    self.targetEnvironments = targetEnvironments
+    self.targetRuntimes = targetRuntimes
+    self.targetPointerAuthenticationSchemes = targetPointerAuthenticationSchemes
+    self.targetObjectFileFormats = targetObjectFileFormats
+    self.targetPointerBitWidth = targetPointerBitWidth
+    self.targetAtomicBitWidths = targetAtomicBitWidths
+    self.endianness = endianness
+    self.languageVersion = languageVersion
+    self.compilerVersion = compilerVersion
+  }
+
+  /// The set of custom conditions that are present and can be used with `#if`.
+  ///
+  /// Custom build conditions can be set by the `-D` command line option to
+  /// the Swift compiler. For example, `-DDEBUG` sets the custom condition
+  /// named `DEBUG`, which could be checked with, e.g.,
+  ///
+  /// ```swift
+  /// #if DEBUG
+  /// // ...
+  /// #endif
+  /// ```
+  public var customConditions: Set<String> = []
+
+  /// The set of language features that are enabled.
+  ///
+  /// Features are determined by the Swift compiler, language mode, and other
+  /// options such as `--enable-upcoming-feature`, and can be checked with
+  /// the `hasFeature` syntax, e.g.,
+  ///
+  /// ```swift
+  /// #if hasFeature(VariadicGenerics)
+  /// // ...
+  /// #endif
+  /// ```
+  public var features: Set<String> = []
+
+  /// The set of attributes that are available.
+  ///
+  /// Attributes are determined by the Swift compiler. They can be checked
+  /// with `hasAttribute` syntax, e.g.,
+  ///
+  /// ```swift
+  /// #if hasAttribute(available)
+  /// // ...
+  /// #endif
+  /// ```
+  public var attributes: Set<String> = []
+
+  /// The active target OS names, e.g., "Windows", "iOS".
+  public var targetOSNames: Set<String> = []
```

Comment:
I really liked the doc comment that links these properties to the clauses in `#if` that can be used to check them in the previous properties, like on `attributes` because it easily explains what the values mean. Do you think you could add equivalent documentation for all the other properties as well?